### ISSUE TITLE
Expected  to be a scalar value but was an array

### ISSUE THREAD

**Issue by superern** [OpenZeppelin/openzeppelin-sdk#1186]
![image](https://user-images.githubusercontent.com/39351808/62920336-0c645b80-bdd8-11e9-80fd-9c475abddb27.png)

I'm trying to do [erc20-onboarding](https://docs.openzeppelin.com/sdk/2.5/erc20-onboarding) but unfortunately, I am not able to create/deploy the UpgradeableToken Contract because of this error when trying to input my answer for "decimal".

> Expected  to be a scalar value but was an array

**Comment by superern** [comment]
I did a temporary quick-fix solution just to create/deploy the contract, I manually commented part of the code to run it but I hope this can be fixed permanently.

> openzeppelin\cli\lib\utils\input.js

![image](https://user-images.githubusercontent.com/39351808/62920526-9f9d9100-bdd8-11e9-98c5-60e3fa5a2b2a.png)

**Comment by spalladino** [comment]
Hey @superern! As a temporary workaround, can you try _without_ passing the `--args` parameter? It should force the CLI to prompt you for the arguments one by one, which may help with this issue. Still, we're flagging this as a bug to take care of this on our end as well.

**Comment by superern** [comment]
@spalladino , yes it worked! unfortunately, there were too many "initialize" function and instead of reading the initialize in my Custom Contract, it reads the "initialize" in ERC20Detailed imported contract.

My problem right now is when I choose one of the "initialize" function, I cannot run the other "initialize" function. I was trying to find a way if there's some sort of running all the "initialize" functions since some of those are required.

`contract NewGenerationToken is Initializable, ERC20Detailed, ERC20Mintable, Ownable {

    uint256 test;
    /**
     * @dev Initialization function.
     * @dev This function will initialize the new upgradeable ERC20 contract and will set up the ERC20 migrator.
     */
    function initialize(ERC20Detailed _legacyToken, ERC20Migrator _migrator) initializer public {
        ERC20Mintable.initialize(address(_migrator));
        ERC20Detailed.initialize(_legacyToken.name(), _legacyToken.symbol(), 18);
    }

}`

This is the snippet of the contract I am using with multiple "initialize" functions.
The inherited contracts that have it are:

- Ownable
- ERC20Detailed
- ERC20Mintable
- Parent contract (NewGenerationToken)

And as you can see, it needs to run other "initialize" functions too.

Thank you for your time looking at the issue!

**Comment by Emobe** [comment]
I have this problem too. Is there no way to specify the initialize function you wish to call yet?

**Comment by spalladino** [comment]
@Emobe if you do not set a `--init` or `--args` option, the CLI will prompt you to choose which function you want to call, allowing you to pick the correct function from a list.

**Comment by robsecord** [comment]
Rename the initialize function your top-level contract:
`function initializeAll(address sender) public initializer { ...`

Then you can call in a shell script:
`contractAddress=$(oz create MyContractName --init initializeAll --args ${ownerAccount} --no-interactive | tail -n 1)`
